crates/fluvio-protocol/src/record/batch.rs

```diff
@@ -67,11 +68,24 @@ pub const BATCH_PREAMBLE_SIZE: usize = size_of::<Offset>()     // Offset
 
 pub const BATCH_FILE_HEADER_SIZE: usize = BATCH_PREAMBLE_SIZE + BATCH_HEADER_SIZE;
 
+#[derive(Clone, Default, Debug, Encoder, PartialEq)]
+pub struct SchemaId(u32);
+
+impl Decoder for SchemaId {
+    fn decode<T: Buf>(&mut self, src: &mut T, version: Version) -> Result<(), Error> {
+        let mut sid: u32 = 0;
+        sid.decode(src, version)?;
+        self.0 = sid;
+        Ok(())
+    }
+}
+
 #[derive(Default, Debug)]
 pub struct Batch<R = MemoryRecords> {
     pub base_offset: Offset,
     pub batch_len: i32, // only for decoding
     pub header: BatchHeader,
+    pub schema_id: Option<SchemaId>,
```

Comment:
need version

Reply:
it might be better to just use SchemaId instead of using option since SchemaId can tell if schema is present

Reply:
Sounds good, the struct in memory is a little more compact w/o the option too.